sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala

```diff
@@ -203,6 +204,32 @@ class HiveSqlAstBuilder extends SparkSqlAstBuilder {
   }
 
   /**
+   * Create a [[CreateTableLike]] command.
+   */
+  override def visitCreateTableLike(ctx: CreateTableLikeContext): LogicalPlan = withOrigin(ctx) {
+    // Get the table header.
+    val (table, temp, ifNotExists, external) = visitCreateTableHeader(ctx.createTableHeader)
+    val sourceTable = visitTableIdentifier(ctx.source)
+
+    val tableType = if (external) {
+      throw new ParseException("Unsupported operation: EXTERNAL clause.", ctx)
+    } else {
+      CatalogTableType.MANAGED_TABLE
+    }
+
+    // Unsupported clauses.
+    if (temp) {
+      throw new ParseException(s"Unsupported operation: TEMPORARY clause.", ctx)
+    }
+    val tableDesc = CatalogTable(
+      identifier = table,
+      tableType = tableType,
+      schema = Seq.empty,
+      storage = EmptyStorageFormat)
+    CreateTableLike(tableDesc, ifNotExists, sourceTable)
```

Comment:
why pass in this incomplete `tableDesc`? We should just pass in `name` and `sourceName`, and call `catalog.getTableMetadata(sourceName)` in the DDL command.